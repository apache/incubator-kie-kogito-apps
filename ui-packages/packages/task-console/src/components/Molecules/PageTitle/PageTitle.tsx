import React from 'react';
import { Flex, FlexItem, Title } from '@patternfly/react-core';
import { componentOuiaProps, OUIAProps } from '@kogito-apps/ouia-tools';

export interface IOwnProps {
  title: any;
  extra?: JSX.Element;
}

const PageTitle: React.FC<IOwnProps & OUIAProps> = ({
  title,
  extra,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <React.Fragment>
      <Flex {...componentOuiaProps(ouiaId, 'page-title', ouiaSafe)}>
        <FlexItem spacer={{ default: 'spacerSm' }}>
          <Title headingLevel="h1" size="4xl">
            {title}
          </Title>
        </FlexItem>
        {extra ? (
          <FlexItem spacer={{ default: 'spacerSm' }}>{extra}</FlexItem>
        ) : null}
      </Flex>
    </React.Fragment>
  );
};

export default PageTitle;
