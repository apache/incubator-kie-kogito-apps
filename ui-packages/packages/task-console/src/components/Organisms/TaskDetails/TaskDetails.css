.kogito-task-console-task-details-u-WordBreak {
  word-break: break-all;
}