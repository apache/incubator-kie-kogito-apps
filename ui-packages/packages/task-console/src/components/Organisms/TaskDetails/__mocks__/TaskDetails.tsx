import React from 'react';

const MockedTaskDetails = (): React.ReactElement => {
  return <></>;
};

export default MockedTaskDetails;
