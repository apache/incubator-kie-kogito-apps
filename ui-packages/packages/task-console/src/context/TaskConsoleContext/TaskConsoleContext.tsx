import React from 'react';

export interface IContext<T> {
  setActiveItem(item: any);
  getActiveItem(): any;
}

export class DefaultContext<T> implements IContext<T> {
  private item: T;

  getActiveItem(): T {
    return this.item;
  }

  setActiveItem(item: T) {
    this.item = item;
  }
}

const TaskConsoleContext = React.createContext<IContext<any>>(
  new DefaultContext()
);

export default TaskConsoleContext;
