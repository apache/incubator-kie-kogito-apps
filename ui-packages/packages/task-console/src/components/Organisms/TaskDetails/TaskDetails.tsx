/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import Moment from 'react-moment';
import {
  Form,
  FormGroup,
  Text,
  TextVariants,
  Tooltip
} from '@patternfly/react-core';
import React, { useContext, useState } from 'react';
import {
  componentOuiaProps,
  EndpointLink,
  GraphQL,
  KogitoEmptyState,
  KogitoEmptyStateType,
  OUIAProps
} from '@kogito-apps/common';
import TaskConsoleContext, {
  IContext
} from '../../../context/TaskConsoleContext/TaskConsoleContext';
import TaskState from '../../Atoms/TaskState/TaskState';
import { resolveTaskPriority, trimTaskEndpoint } from '../../../util/Utils';
import './TaskDetails.css';
import UserTaskInstance = GraphQL.UserTaskInstance;

interface IOwnProps {
  userTaskInstance?: UserTaskInstance;
}

const TaskDetails: React.FC<IOwnProps & OUIAProps> = ({
  userTaskInstance,
  ouiaId,
  ouiaSafe
}) => {
  const context: IContext<UserTaskInstance> = useContext(TaskConsoleContext);
  const [userTask, setUserTask] = useState<UserTaskInstance>();

  if (!userTask) {
    if (userTaskInstance) {
      setUserTask(userTaskInstance);
    } else {
      if (context.getActiveItem()) {
        setUserTask(context.getActiveItem());
      }
    }
  }

  if (!userTask) {
    return (
      <KogitoEmptyState
        type={KogitoEmptyStateType.Info}
        title="Cannot show details"
        body="Unable to show details for empty task"
      />
    );
  }

  return (
    <React.Fragment>
      <div
        {...componentOuiaProps(
          ouiaId ? ouiaId : userTask.id,
          'task-details',
          ouiaSafe
        )}
      >
        <Form>
          <FormGroup label="Name" fieldId="name">
            <Text component={TextVariants.p}>{userTask.referenceName}</Text>
          </FormGroup>
          {userTask.description && (
            <FormGroup label="Description" fieldId="description">
              <Text component={TextVariants.p}>{userTask.description}</Text>
            </FormGroup>
          )}
          <FormGroup label="ID" fieldId="id">
            <Text component={TextVariants.p}>{userTask.id}</Text>
          </FormGroup>
          <FormGroup label="State" fieldId="state">
            <TaskState task={userTask} />
          </FormGroup>
          <FormGroup label="Priority" fieldId="priority">
            <Text component={TextVariants.p}>
              {resolveTaskPriority(userTask.priority)}
            </Text>
          </FormGroup>
          <FormGroup label="Owner" fieldId="owner">
            <Text component={TextVariants.p}>
              {userTask.actualOwner || '-'}
            </Text>
          </FormGroup>
          <FormGroup label="Started" fieldId="started">
            {userTask.started ? (
              <Text component={TextVariants.p}>
                <Moment fromNow>{new Date(`${userTask.started}`)}</Moment>
              </Text>
            ) : (
              '-'
            )}
          </FormGroup>
          {userTask.completed && (
            <FormGroup label="Completed" fieldId="completed">
              <Text component={TextVariants.p}>
                <Moment fromNow>{new Date(`${userTask.completed}`)}</Moment>
              </Text>
            </FormGroup>
          )}
          <FormGroup label="Process" fieldId="process">
            <Text component={TextVariants.p}>{userTask.processId}</Text>
          </FormGroup>
          <FormGroup label="Process Instance ID" fieldId="processInstance">
            <Text component={TextVariants.p}>{userTask.processInstanceId}</Text>
          </FormGroup>
          <FormGroup label="Endpoint" fieldId="endpoint">
            <Tooltip content={userTask.endpoint}>
              <EndpointLink
                serviceUrl={userTask.endpoint}
                linkLabel={trimTaskEndpoint(userTask)}
                isLinkShown={false}
              />
            </Tooltip>
          </FormGroup>
        </Form>
      </div>
    </React.Fragment>
  );
};

export default TaskDetails;
