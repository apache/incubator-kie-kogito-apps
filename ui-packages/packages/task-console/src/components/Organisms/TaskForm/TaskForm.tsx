/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useContext, useEffect, useState } from 'react';
import axios from 'axios';
import { Stack, StackItem } from '@patternfly/react-core';
import {
  GraphQL,
  KogitoEmptyState,
  KogitoEmptyStateType,
  KogitoSpinner
} from '@kogito-apps/common';
import TaskConsoleContext, {
  IContext
} from '../../../context/TaskConsoleContext/TaskConsoleContext';
import FormRenderer from '../../Molecules/FormRenderer/FormRenderer';
import { TaskFormSubmitHandler } from '../../../util/uniforms/TaskFormSubmitHandler/TaskFormSubmitHandler';
import { FormSchema } from '../../../util/uniforms/FormSchema';
import { getTaskSchemaEndPoint } from '../../../util/Utils';
import FormNotification, {
  Notification
} from '../../Atoms/FormNotification/FormNotification';
import UserTaskInstance = GraphQL.UserTaskInstance;
import { NotificationType } from '../../../util/Variants';

interface IOwnProps {
  userTaskInstance?: UserTaskInstance;
  successCallback?: () => void;
  errorCallback?: () => void;
}

const TaskForm: React.FC<IOwnProps> = ({
  userTaskInstance,
  successCallback,
  errorCallback
}) => {
  // tslint:disable: no-floating-promises
  const context: IContext<UserTaskInstance> = useContext(TaskConsoleContext);
  const [notification, setNotification] = useState<Notification>();
  const [loading, setLoading] = useState<boolean>(true);
  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
  const [submitted, setSubmitted] = useState<boolean>(false);
  const [stateUserTask, setStateUserTask] = useState<UserTaskInstance>();
  const [taskFormSchema, setTaskFormSchema] = useState<FormSchema>(null);

  if (!stateUserTask) {
    if (userTaskInstance) {
      setStateUserTask(userTaskInstance);
    } else {
      if (context.getActiveItem()) {
        setStateUserTask(context.getActiveItem());
      }
    }
  }

  useEffect(() => {
    loadForm();
  }, [stateUserTask]);

  const loadForm = () => {
    if (stateUserTask) {
      const endpoint = getTaskSchemaEndPoint(stateUserTask, context.getUser());

      axios
        .get(endpoint, {
          headers: {
            'Content-Type': 'application/json',
            Accept: 'application/json',
            crossorigin: 'true',
            'Access-Control-Allow-Origin': '*'
          }
        })
        .then(res => {
          if (res.status === 200) {
            setTaskFormSchema(res.data);
          }
          setLoading(false);
        })
        .catch(er => {
          setLoading(false);
        });
    }
  };

  const showNotification = (
    notificationType: NotificationType,
    submitMessage: string,
    submitCallback?: () => void,
    notificationDetails?: string
  ) => {
    setNotification({
      type: notificationType,
      message: submitMessage,
      details: notificationDetails,
      customAction: submitCallback
        ? {
            label: 'Select another Task',
            onClick: () => {
              setNotification(null);
              if (submitCallback) {
                submitCallback();
              }
            }
          }
        : undefined,
      close: () => {
        setNotification(null);
      }
    });
  };

  if (stateUserTask) {
    if (loading) {
      return (
        <KogitoSpinner
          spinnerText={'Loading form for task: ' + stateUserTask.name}
        />
      );
    }

    if (isSubmitting) {
      return <KogitoSpinner spinnerText={'Submitting form...'} />;
    }

    if (taskFormSchema) {
      const notifySuccess = (phase: string) => {
        const message = `Task '${userTaskInstance.referenceName}' successfully transitioned to phase '${phase}'.`;

        showNotification(NotificationType.SUCCESS, message, successCallback);
        setIsSubmitting(false);
        setSubmitted(true);
      };

      const notifyError = (phase: string, error?: string) => {
        const message = `Task '${userTaskInstance.referenceName}' couldn't transition to phase '${phase}'.`;

        showNotification(NotificationType.ERROR, message, errorCallback, error);
        setIsSubmitting(false);
      };

      const formSubmitHandler = new TaskFormSubmitHandler(
        stateUserTask,
        taskFormSchema,
        context.getUser(),
        () => setIsSubmitting(true),
        phase => notifySuccess(phase),
        (phase, errorMessage) => notifyError(phase, errorMessage)
      );

      const formData = JSON.parse(stateUserTask.inputs);

      return (
        <Stack hasGutter>
          {notification && (
            <StackItem>
              <FormNotification notification={notification} />
            </StackItem>
          )}
          <StackItem>
            <FormRenderer
              formSchema={taskFormSchema}
              model={formData}
              readOnly={submitted}
              formSubmitHandler={formSubmitHandler}
            />
          </StackItem>
        </Stack>
      );
    }
  }

  return (
    <KogitoEmptyState
      type={KogitoEmptyStateType.Info}
      title="No form to show"
      body="Cannot find form"
    />
  );
};

export default TaskForm;
