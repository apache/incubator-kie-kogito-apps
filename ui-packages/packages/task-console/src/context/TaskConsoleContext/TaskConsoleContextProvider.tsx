import React from 'react';
import TaskConsoleContext, { DefaultContext } from './TaskConsoleContext';
import { TaskInfo } from '../../model/TaskInfo';

const TaskConsoleContextProvider: React.FC = props => {
  return (
    <TaskConsoleContext.Provider value={new DefaultContext<TaskInfo>()}>
      {props.children}
    </TaskConsoleContext.Provider>
  );
};

export default TaskConsoleContextProvider;
