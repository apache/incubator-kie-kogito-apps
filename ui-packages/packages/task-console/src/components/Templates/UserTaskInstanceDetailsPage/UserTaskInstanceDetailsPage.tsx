/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useContext, useEffect, useState } from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbItem,
  Card,
  CardBody,
  CardHeader,
  Grid,
  GridItem,
  PageSection,
  Title,
  Text,
  TextVariants
} from '@patternfly/react-core';
import {
  componentOuiaProps,
  GraphQL,
  KogitoEmptyState,
  KogitoEmptyStateType,
  ouiaPageTypeAndObjectId,
  OUIAProps
} from '@kogito-apps/common';
import TaskConsoleContext, {
  IContext
} from '../../../context/TaskConsoleContext/TaskConsoleContext';
import PageTitle from '../../Molecules/PageTitle/PageTitle';
import TaskState from '../../Atoms/TaskState/TaskState';
import TaskForm from '../../Organisms/TaskForm/TaskForm';
import { TaskStateType } from '../../../util/Variants';
import UserTaskInstance = GraphQL.UserTaskInstance;
import TaskDetails from '../../Organisms/TaskDetails/TaskDetails';

interface MatchProps {
  taskId: string;
}

const UserTaskInstanceDetailsPage: React.FC<RouteComponentProps<
  MatchProps,
  {},
  {}
> &
  OUIAProps> = ({ ouiaId, ouiaSafe, ...props }) => {
  const id = props.match.params.taskId;

  const [userTask, setUserTask] = useState<UserTaskInstance>();

  const context: IContext<UserTaskInstance> = useContext(TaskConsoleContext);

  useEffect(() => {
    window.onpopstate = () => {
      props.history.push({ state: { ...props.location.state } });
    };
  });

  useEffect(() => {
    return ouiaPageTypeAndObjectId('user-tasks', id);
  });

  useEffect(() => {
    if (context.getActiveItem()) {
      setUserTask(context.getActiveItem());
    }
  }, []);

  if (!userTask) {
    return (
      <KogitoEmptyState
        type={KogitoEmptyStateType.Info}
        title={'Cannot find task'}
        body={`Cannot find task with id '${id}'`}
      />
    );
  }

  return (
    <React.Fragment>
      <div {...componentOuiaProps(ouiaId, 'UserTaskInstanceDetails', ouiaSafe)}>
        <PageSection variant="light">
          <Breadcrumb>
            <BreadcrumbItem>
              <Link to={'/'}>Task Inbox</Link>
            </BreadcrumbItem>
            <BreadcrumbItem>{userTask.referenceName}</BreadcrumbItem>
          </Breadcrumb>

          <PageTitle
            title={userTask.referenceName}
            extra={<TaskState task={userTask} variant={TaskStateType.LABEL} />}
          />

          <Text component={TextVariants.small}>ID: {userTask.id}</Text>
        </PageSection>
        <PageSection>
          <Grid hasGutter md={1} className="pf-u-h-100">
            <GridItem span={8} className="pf-u-h-100">
              <Card className="pf-u-h-100">
                <CardHeader>
                  <Title headingLevel="h3" size="xl">
                    Form
                  </Title>
                </CardHeader>
                <CardBody className="pf-u-h-100">
                  <TaskForm
                    userTaskInstance={userTask}
                    successCallback={() => props.history.push('/')}
                    errorCallback={() => props.history.push('/')}
                  />
                </CardBody>
              </Card>
            </GridItem>
            <GridItem span={4} className="pf-u-h-100">
              <Card className="pf-u-h-100">
                <CardHeader>
                  <Title headingLevel="h3" size="xl">
                    Details
                  </Title>
                </CardHeader>
                <CardBody className="pf-u-h-100">
                  <TaskDetails userTaskInstance={userTask} />
                </CardBody>
              </Card>
            </GridItem>
          </Grid>
        </PageSection>
      </div>
    </React.Fragment>
  );
};

export default UserTaskInstanceDetailsPage;
