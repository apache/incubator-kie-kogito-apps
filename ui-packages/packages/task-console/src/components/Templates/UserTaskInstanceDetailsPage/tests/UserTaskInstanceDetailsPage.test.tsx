/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import * as H from 'history';
import {
  DefaultUser,
  getWrapper,
  GraphQL,
  KogitoEmptyState,
  User
} from '@kogito-apps/common';
import UserTaskInstance = GraphQL.UserTaskInstance;
import TaskConsoleContext, {
  DefaultContext
} from '../../../../context/TaskConsoleContext/TaskConsoleContext';
import TaskConsoleContextProvider from '../../../../context/TaskConsoleContext/TaskConsoleContextProvider';
import UserTaskInstanceDetailsPage from '../UserTaskInstanceDetailsPage';
import { BrowserRouter } from 'react-router-dom';
import { Breadcrumb, Text } from '@patternfly/react-core';
import PageTitle from '../../../Molecules/PageTitle/PageTitle';
import TaskForm from '../../../Organisms/TaskForm/TaskForm';
import TaskDetails from '../../../Organisms/TaskDetails/TaskDetails';

const MockedComponent = (): React.ReactElement => {
  return <></>;
};

jest.mock('../../../Atoms/TaskState/TaskState');
jest.mock('../../../Molecules/PageTitle/PageTitle');
jest.mock('../../../Organisms/TaskDetails/TaskDetails');
jest.mock('../../../Organisms/TaskForm/TaskForm');

jest.mock('@kogito-apps/common', () => ({
  ...jest.requireActual('@kogito-apps/common'),
  KogitoEmptyState: () => {
    return <MockedComponent />;
  }
}));

jest.mock('@patternfly/react-core', () => ({
  ...jest.requireActual('@patternfly/react-core'),
  Breadcrumb: () => {
    return <MockedComponent />;
  },
  BreadcrumbItem: () => {
    return <MockedComponent />;
  }
}));

const userTaskInstance: UserTaskInstance = {
  id: '45a73767-5da3-49bf-9c40-d533c3e77ef3',
  description: null,
  name: 'VisaApplication',
  referenceName: 'Apply for visa',
  priority: '1',
  processInstanceId: '9ae7ce3b-d49c-4f35-b843-8ac3d22fa427',
  processId: 'travels',
  rootProcessInstanceId: null,
  rootProcessId: null,
  state: 'Ready',
  actualOwner: null,
  adminGroups: [],
  adminUsers: [],
  completed: null,
  started: '2020-02-19T11:11:56.282Z',
  excludedUsers: [],
  potentialGroups: [],
  potentialUsers: [],
  inputs:
    '{"Skippable":"true","trip":{"city":"Boston","country":"US","begin":"2020-02-19T23:00:00.000+01:00","end":"2020-02-26T23:00:00.000+01:00","visaRequired":true},"TaskName":"VisaApplication","NodeName":"Apply for visa","traveller":{"firstName":"Rachel","lastName":"White","email":"rwhite@gorle.com","nationality":"Polish","address":{"street":"Cabalone","city":"Zerf","zipCode":"765756","country":"Poland"}},"Priority":"1"}',
  outputs: '{}',
  lastUpdate: '2020-02-19T11:11:56.282Z',
  endpoint:
    'http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3'
};

const testUser: User = new DefaultUser('test', ['group1', 'group2']);

const props = {
  match: {
    params: {
      taskId: '45a73767-5da3-49bf-9c40-d533c3e77ef3'
    },
    url: '',
    isExact: false,
    path: ''
  },
  location: H.createLocation(''),
  history: H.createBrowserHistory()
};

describe('UserTaskInstanceDetailsPage tests', () => {
  it('Test empty state', () => {
    const wrapper = getWrapper(
      <TaskConsoleContextProvider user={testUser}>
        <UserTaskInstanceDetailsPage {...props} />
      </TaskConsoleContextProvider>,
      'UserTaskInstanceDetailsPage'
    );

    expect(wrapper).toMatchSnapshot();

    const emptyState = wrapper.find(KogitoEmptyState);
    expect(emptyState.exists()).toBeTruthy();

    expect(emptyState.props().body).toStrictEqual(
      "Cannot find task with id '45a73767-5da3-49bf-9c40-d533c3e77ef3'"
    );
    expect(emptyState.props().title).toStrictEqual('Cannot find task');
  });

  it('Test active task', () => {
    const context = new DefaultContext<UserTaskInstance>(testUser);

    context.setActiveItem(userTaskInstance);

    const wrapper = getWrapper(
      <TaskConsoleContext.Provider value={context}>
        <BrowserRouter>
          <UserTaskInstanceDetailsPage {...props} />
        </BrowserRouter>
      </TaskConsoleContext.Provider>,
      'UserTaskInstanceDetailsPage'
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(Breadcrumb).exists()).toBeTruthy();

    const title = wrapper.find(PageTitle);

    expect(title.exists()).toBeTruthy();

    expect(title.props().title).toStrictEqual(userTaskInstance.referenceName);
    expect(title.props().extra).not.toBeNull();

    const id = wrapper.find(Text);
    expect(id.exists()).toBeTruthy();
    expect(id.html()).toContain(`ID: ${userTaskInstance.id}`);

    const taskForm = wrapper.find(TaskForm);

    expect(taskForm.exists()).toBeTruthy();
    expect(taskForm.props().userTaskInstance).toStrictEqual(userTaskInstance);
    expect(taskForm.props().successCallback).not.toBeNull();
    expect(taskForm.props().errorCallback).not.toBeNull();

    const taskDetails = wrapper.find(TaskDetails);

    expect(taskDetails.exists()).toBeTruthy();
    expect(taskDetails.props().userTaskInstance).toStrictEqual(
      userTaskInstance
    );
  });
});
