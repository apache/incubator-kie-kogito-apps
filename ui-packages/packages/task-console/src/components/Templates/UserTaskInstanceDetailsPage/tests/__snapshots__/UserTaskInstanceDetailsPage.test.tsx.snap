// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTaskInstanceDetailsPage tests Test active task 1`] = `
<UserTaskInstanceDetailsPage
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": false,
      "params": Object {
        "taskId": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
      },
      "path": "",
      "url": "",
    }
  }
>
  <div
    data-ouia-component-type="UserTaskInstanceDetails"
    data-ouia-safe={true}
  >
    <PageSection
      variant="light"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
      >
        <Breadcrumb>
          <MockedComponent />
        </Breadcrumb>
        <MockedPageTitle
          extra={
            <MockedTaskState
              task={
                Object {
                  "actualOwner": null,
                  "adminGroups": Array [],
                  "adminUsers": Array [],
                  "completed": null,
                  "description": null,
                  "endpoint": "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
                  "excludedUsers": Array [],
                  "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
                  "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}",
                  "lastUpdate": "2020-02-19T11:11:56.282Z",
                  "name": "VisaApplication",
                  "outputs": "{}",
                  "potentialGroups": Array [],
                  "potentialUsers": Array [],
                  "priority": "1",
                  "processId": "travels",
                  "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
                  "referenceName": "Apply for visa",
                  "rootProcessId": null,
                  "rootProcessInstanceId": null,
                  "started": "2020-02-19T11:11:56.282Z",
                  "state": "Ready",
                }
              }
              variant="label"
            />
          }
          title="Apply for visa"
        />
        <Text
          component="small"
        >
          <small
            className=""
            data-pf-content={true}
          >
            ID: 
            45a73767-5da3-49bf-9c40-d533c3e77ef3
          </small>
        </Text>
      </section>
    </PageSection>
    <PageSection>
      <section
        className="pf-c-page__main-section"
      >
        <Grid
          className="pf-u-h-100"
          hasGutter={true}
          md={1}
        >
          <div
            className="pf-l-grid pf-m-all-1-col-on-md pf-m-gutter pf-u-h-100"
          >
            <GridItem
              className="pf-u-h-100"
              span={8}
            >
              <div
                className="pf-l-grid__item pf-m-8-col pf-u-h-100"
              >
                <Card
                  className="pf-u-h-100"
                >
                  <article
                    className="pf-c-card pf-u-h-100"
                    data-ouia-component-id={0}
                    data-ouia-component-type="PF4/Card"
                    data-ouia-safe={true}
                  >
                    <CardHeader>
                      <div
                        className="pf-c-card__header"
                      >
                        <Title
                          headingLevel="h3"
                          size="xl"
                        >
                          <h3
                            className="pf-c-title pf-m-xl"
                          >
                            Form
                          </h3>
                        </Title>
                      </div>
                    </CardHeader>
                    <CardBody
                      className="pf-u-h-100"
                    >
                      <div
                        className="pf-c-card__body pf-u-h-100"
                      >
                        <MockedTaskForm
                          errorCallback={[Function]}
                          successCallback={[Function]}
                          userTaskInstance={
                            Object {
                              "actualOwner": null,
                              "adminGroups": Array [],
                              "adminUsers": Array [],
                              "completed": null,
                              "description": null,
                              "endpoint": "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
                              "excludedUsers": Array [],
                              "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
                              "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}",
                              "lastUpdate": "2020-02-19T11:11:56.282Z",
                              "name": "VisaApplication",
                              "outputs": "{}",
                              "potentialGroups": Array [],
                              "potentialUsers": Array [],
                              "priority": "1",
                              "processId": "travels",
                              "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
                              "referenceName": "Apply for visa",
                              "rootProcessId": null,
                              "rootProcessInstanceId": null,
                              "started": "2020-02-19T11:11:56.282Z",
                              "state": "Ready",
                            }
                          }
                        />
                      </div>
                    </CardBody>
                  </article>
                </Card>
              </div>
            </GridItem>
            <GridItem
              className="pf-u-h-100"
              span={4}
            >
              <div
                className="pf-l-grid__item pf-m-4-col pf-u-h-100"
              >
                <Card
                  className="pf-u-h-100"
                >
                  <article
                    className="pf-c-card pf-u-h-100"
                    data-ouia-component-id={1}
                    data-ouia-component-type="PF4/Card"
                    data-ouia-safe={true}
                  >
                    <CardHeader>
                      <div
                        className="pf-c-card__header"
                      >
                        <Title
                          headingLevel="h3"
                          size="xl"
                        >
                          <h3
                            className="pf-c-title pf-m-xl"
                          >
                            Details
                          </h3>
                        </Title>
                      </div>
                    </CardHeader>
                    <CardBody
                      className="pf-u-h-100"
                    >
                      <div
                        className="pf-c-card__body pf-u-h-100"
                      >
                        <MockedTaskDetails
                          userTaskInstance={
                            Object {
                              "actualOwner": null,
                              "adminGroups": Array [],
                              "adminUsers": Array [],
                              "completed": null,
                              "description": null,
                              "endpoint": "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
                              "excludedUsers": Array [],
                              "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
                              "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}",
                              "lastUpdate": "2020-02-19T11:11:56.282Z",
                              "name": "VisaApplication",
                              "outputs": "{}",
                              "potentialGroups": Array [],
                              "potentialUsers": Array [],
                              "priority": "1",
                              "processId": "travels",
                              "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
                              "referenceName": "Apply for visa",
                              "rootProcessId": null,
                              "rootProcessInstanceId": null,
                              "started": "2020-02-19T11:11:56.282Z",
                              "state": "Ready",
                            }
                          }
                        />
                      </div>
                    </CardBody>
                  </article>
                </Card>
              </div>
            </GridItem>
          </div>
        </Grid>
      </section>
    </PageSection>
  </div>
</UserTaskInstanceDetailsPage>
`;

exports[`UserTaskInstanceDetailsPage tests Test empty state 1`] = `
<UserTaskInstanceDetailsPage
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": false,
      "params": Object {
        "taskId": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
      },
      "path": "",
      "url": "",
    }
  }
>
  <KogitoEmptyState
    body="Cannot find task with id '45a73767-5da3-49bf-9c40-d533c3e77ef3'"
    title="Cannot find task"
    type={3}
  >
    <MockedComponent />
  </KogitoEmptyState>
</UserTaskInstanceDetailsPage>
`;
